Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx

```diff
@@ -297,23 +312,33 @@ class DataDecoderTask
       }
     }
 
-    const size_t OUT_SIZE = sizeof(o2::mch::Digit) * digits.size();
-
     // send the output buffer via DPL
-    char* outbuffer = nullptr;
-    outbuffer = (char*)realloc(outbuffer, OUT_SIZE);
-    memcpy(outbuffer, digits.data(), OUT_SIZE);
+    const size_t digitsSize = sizeof(o2::mch::Digit) * digits.size();
```

Comment:
the two pieces of code below are just repetitions, could be extracted to a (templatized) function, e.g. `char* createBuffer(T container)` ?

Reply:
Done.